includes/CMB2_Utils.php

```diff
@@ -76,6 +76,99 @@ public static function image_id_from_url( $img_url ) {
 	}
 
 	/**
+	 * Utility method to get a combined list of default and custom registered image sizes
+	 * @since  2.x.x.x
+	 * @link   http://core.trac.wordpress.org/ticket/18947
+	 * @global array $_wp_additional_image_sizes
+	 * @return array $image_sizes The image sizes
+	 */
+	static function get_available_image_sizes() {
+		global $_wp_additional_image_sizes;
+
+		$default_image_sizes = array( 'thumbnail', 'medium', 'large' );
+		foreach ( $default_image_sizes as $size ) {
+			$image_sizes[ $size ] = array(
+				'height' => intval( get_option( "{$size}_size_h" ) ),
+				'width'  => intval( get_option( "{$size}_size_w" ) ),
+				'crop'   => get_option( "{$size}_crop" ) ? get_option( "{$size}_crop" ) : false,
+			);
+		}
+
+		if ( isset( $_wp_additional_image_sizes ) && count( $_wp_additional_image_sizes ) ) {
+			$image_sizes = array_merge( $image_sizes, $_wp_additional_image_sizes );
+		}
+
+		return $image_sizes;
+	}
+
+	/**
+	 * Utility method to return the closest named size from an array of values
+	 *
+	 * Based off of WordPress's image_get_intermediate_size()
+	 * If the size matches an existing size then it will be used. If there is no
+	 * direct match, then the nearest image size larger than the specified size
+	 * will be used. If nothing is found, then the function will return false.
+	 * Uses get_available_image_sizes() to get all available sizes.
+	 *
+	 * @since  2.x.x.x
+	 * @param  array|string $size Image size. Accepts an array of width and height (in that order).
+	 * @return false|string $data Named image size e.g. 'thumbnail'.
+	 */
+	public static function get_named_size( $size ) {
+		$image_sizes = self::get_available_image_sizes();
```

Comment:
i'd move this inside the `if ( is_array( $size ) ) {` condition since it's not needed otherwise.